protocol/src/test/java/io/camunda/zeebe/protocol/ZbColumnFamiliesTest.java

```diff
@@ -30,6 +31,101 @@ void shouldNotReuseEnumValues() {
         .doesNotHaveDuplicates();
   }
 
+  /**
+   * This test is to ensure that the enum values aren't accidentally changed when adding or
+   * backporting new entries.
+   */
+  @Test
+  void shouldHaveExplicitEnumValues() {
+    assertThat(Arrays.stream(ZbColumnFamilies.values()))
+        .extracting(ZbColumnFamilies::name, ZbColumnFamilies::getValue)
+        .containsExactly(
+            tuple("DEFAULT", 0),
+            tuple("KEY", 1),
+            tuple("DEPRECATED_PROCESS_VERSION", 2),
+            tuple("DEPRECATED_PROCESS_CACHE", 3),
```

Comment:
I was actually thinking about just checking against the ordinals and that it increments by one, because it would fail only if something was configured wrongly. Here it would now fail all the time, right? Might make people work around such or annoyed by this test, but I guess we will see. If you prefer this way we can also take this.

Reply:
That's a fair point